js/piwik.js

```diff
@@ -7130,6 +7155,125 @@ if (typeof window.Piwik !== 'object') {
+                configRequireConsent = false;
+                var i, requestType;
+                for (i = 0; i < consentRequestsQueue.length; i++) {
+                    requestType = typeof consentRequestsQueue[i];
+                    if (requestType === 'string') {
+                        sendRequest(consentRequestsQueue[i], configTrackerPause);
+                    } else if (requestType === 'object') {
+                        sendBulkRequest(consentRequestsQueue[i], configTrackerPause);
+                    }
+                }
+                consentRequestsQueue = [];
+            };
+
+            /**
+             * Calling this method will remember that the user has given consent across multiple requests by setting
+             * a cookie. You can optionally define the lifetime of that cookie in milliseconds using a parameter.
+             *
+             * When you call this method, we imply that the user has given consent for this page view, and will also
+             * imply consent for all future page views unless the cookie expires (if timeout defined) or the user
+             * deletes all her or his cookies. This means even if you call {@link requireConsent()}, then all requests
+             * will still be tracked.
+             *
+             * Please note that this feature requires you to set the `cookieDomain` and `cookiePath` correctly and requires
+             * that you do not disable cookies. Please also note that when you call this method, consent will be implied
+             * for all sites that match the configured cookieDomain and cookiePath. Depending on your website structure,
+             * you may need to restrict or widen the scope of the cookie domain/path to ensure the consent is applied
+             * to the sites you want.
+             */
+            this.rememberConsentGiven = function (msToExpire) {
+                if (configCookiesDisabled) {
+                    logConsoleError('rememberConsentGiven is called but cookies are disabled, consent will not be remembered');
+                    return;
+                }
+                this.setConsentGiven();
+                var now = new Date().getTime();
+                setCookie('consent', now, msToExpire, configCookiePath, configCookieDomain, configCookieIsSecure);
+            };
+
+            /**
+             * Calling this method will remove any previously given consent and during this page view no request
+             * will be sent anymore ({@link requireConsent()}) will be called automatically to ensure the removed
+             * consent will be enforced. You may call this method if the user removes consent manually, or if you
+             * want to re-ask for consent after a specific time period.
+             */
+            this.forgetConsentGiven = function () {
+                deleteCookie('consent', configCookiePath, configCookieDomain);
+                this.requireConsent();
```

Comment:
Say you have a checkbox on your homepage where people can configure whether they want to be tracked or not.

Now when a user enables the checkbox, a developer might decide to call `rememberConsentGiven()`. So this way Matomo will always know that the user has explicitly given consent. When the user now later goes to the same page again and disables the checkbox, the developer will need to call `forgetConsentGiven()` to make sure Matomo will no longer assume any consent exists. Does it make it a bit more clear?

Reply:
Oh, I got that part (ie, the `deleteCookie()` line), I didn't get why `this.requireConsent()` had to be called. I assume it would be called in the user's site before `forgetConsentGiven()`?